app/src/main/java/mozilla/lockbox/view/FingerprintAuthDialogFragment.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.view
+
+import android.os.Bundle
+import android.support.v4.app.DialogFragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+import kotlinx.android.synthetic.main.fragment_fingerprint_dialog.view.*
+import mozilla.lockbox.R
+import mozilla.lockbox.presenter.FingerprintDialogPresenter
+import mozilla.lockbox.presenter.FingerprintDialogView
+
+class FingerprintAuthDialogFragment : DialogFragment(), FingerprintDialogView {
+    private lateinit var presenter: FingerprintDialogPresenter
+    private val onAuth = PublishSubject.create<Unit>()
+    private val onError = PublishSubject.create<Unit>()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        presenter = FingerprintDialogPresenter(this)
+        return inflater.inflate(R.layout.fragment_fingerprint_dialog, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        presenter.onViewReady()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        view!!.imageView.setImageResource(R.drawable.ic_fingerprint)
```

Comment:
The need to reset this here will likely go away if you use a `PublishSubject` instead of a `ReplaySubject` for the authentication state.